src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java

```diff
@@ -26,15 +26,27 @@ public boolean isFull() {
 	public int top() {
 		if (isEmpty())
 			return -1;
-		return last.value;
+		return getLast().getValue();
 	}
 
 	public int pop() {
 		if (isEmpty())
 			return -1;
-		int ret = last.value;
-		last = last.prev;
+		int ret = getLast().getValue();
+		setLast(getLast().getPrev());
```

Comment:
This is an exaggeration. Using private fields internally usually does not require encapsulation.